server/src/main/java/org/opensearch/action/search/SearchTaskRequestOperationsListener.java

```diff
@@ -25,6 +25,9 @@ public SearchTaskRequestOperationsListener(TaskResourceTrackingService taskResou
 
     @Override
     public void onRequestEnd(SearchPhaseContext context, SearchRequestContext searchRequestContext) {
+        // Refresh the coordinator node level resource usages
         taskResourceTrackingService.refreshResourceStats(context.getTask());
+        // Remove the shard level resource usages from thread context
+        taskResourceTrackingService.removeTaskResourceUsage();
```

Comment:
I think we should also remove these headers as part of failed requests?
For that we will need to extend `onRequestFailure`.